remoting/subsystem/src/main/java/org/jboss/as/remoting/GenericOutboundConnectionResourceDefinition.java

```diff
@@ -40,39 +44,35 @@ class GenericOutboundConnectionResourceDefinition extends AbstractOutboundConnec
 
     static final PathElement ADDRESS = PathElement.pathElement(CommonAttributes.OUTBOUND_CONNECTION);
 
-    public static final SimpleAttributeDefinition URI = new SimpleAttributeDefinitionBuilder(CommonAttributes.URI, ModelType.STRING, false)
+    static final SimpleAttributeDefinition URI = new SimpleAttributeDefinitionBuilder(CommonAttributes.URI, ModelType.STRING, false)
             .setAllowExpression(true).setValidator(new StringLengthValidator(1, Integer.MAX_VALUE, false, true))
             .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES).build();
 
+    static final Collection<AttributeDefinition> ATTRIBUTES = List.of(URI);
 
     GenericOutboundConnectionResourceDefinition() {
+        this(new GenericOutboundConnectionAdd());
+    }
+
+    private GenericOutboundConnectionResourceDefinition(GenericOutboundConnectionAdd addHandler) {
         super(new Parameters(ADDRESS, RemotingExtension.getResourceDescriptionResolver(CommonAttributes.OUTBOUND_CONNECTION))
-                .setAddHandler(GenericOutboundConnectionAdd.INSTANCE)
-                .setRemoveHandler(new ServiceRemoveStepHandler(OUTBOUND_CONNECTION_CAPABILITY.getCapabilityServiceName(), GenericOutboundConnectionAdd.INSTANCE))
+                .setAddHandler(addHandler)
+                .setRemoveHandler(new ServiceRemoveStepHandler(OUTBOUND_CONNECTION_CAPABILITY.getCapabilityServiceName(), addHandler))
                 .setDeprecatedSince(RemotingSubsystemModel.VERSION_4_0_0.getVersion())
         );
     }
 
     @Override
     public void registerChildren(ManagementResourceRegistration resourceRegistration) {
+        // TODO why does this resource have properties that are never referenced?
```

Comment:
@pferraro @fl4via This dates to:

https://github.com/wildfly/wildfly-core/commit/7281132dee1f595f165720dcd4c91c7ee9dd7256

which came in as part of the big Elytron integration PR https://github.com/wildfly/wildfly-core/pull/2148

due to its incorporation of https://github.com/wildfly-security-incubator/wildfly-core/pull/37

The child is kind of deprecated because this resource is deprecated but I suppose it should be deprecatd on its own. And if we decided to undeprecate this and local-outbound-connection for sure we should deprecate the child.